src/pytesmo/validation_framework/adapters.py

```diff
@@ -502,3 +497,187 @@ def _adapt(self, data: DataFrame) -> DataFrame:
+    2. One or more (minute, s, µs) offset times to be added cumulatively
+
+    -------------------
+    Example input:
+
+         variable    base_time [w.r.t. 2005-02-01]  offset [min]  offset [sec]
+    100  0.889751                            100.0          38.0         999.0
+    101  0.108279                            101.0          40.0        1000.0
+    102 -1.201708                            102.0          39.0         999.0
+
+    Example output:
+
+                         variable
+    2005-05-12 00:55:42  0.889751
+    2005-05-13 00:57:39  0.108279
+    2005-05-14 00:56:38 -1.201708
+
+    Parameters
+    ----------
+     cls: object
+        Reader object, has to have a `read_ts` or `read` method or a method
+        name must be specified in the `read_name` kwarg. The same method will
+        be available for the adapted version of the reader.
+    time_offset_fields: str, list or None
+        name or list of names of the fields that provide information on the time offset.
+        If a list is given, all values will contribute to the offset, assuming that
+        each refers to the previous. For instance:
+        offset = minutes + seconds in the minute + µs in the second
+        NOTE: np.nan values are counted as 0 offset
+        NOTE: if None, no offset is considered
+    time_units: str or list
+        time units that the time_offset_fields are specified in. If a list is given,
+        it should have the same size as the 'time_offset_fields' parameter
+        Can be any of the np.datetime[64] units:
+        https://numpy.org/doc/stable/reference/arrays.datetime.html
+    base_time_field: str, optional. Default is None.
+        If a name is provided, the generic time field will be searched for
+        in the columns; otherwise, it is assumed to be the index
+        NOTE: np.nan values in this field are dropped
+    base_time_reference: str, optional. Default is None.
+        String of format 'YYYY-mm-dd' that can be specified to tranform the
+        'base_time_field' from [units since base_time_reference] to
+        np.datetime[64]. If not provided, it will be assumed that the base_time_field
+        is already in np.datetime[64] units
+    base_time_units: str, optional. Default is "D"
+        Units that the base_time_field is specified in. Only applicable with 'base_time_reference'
+    replace_index: bool, optional. Default is True.
+        If True, the exact timestamp is used as index. Else, it will be added
+        to the dataframe on the column 'output_field'
+    output_field: str, optional. Default is None.
+        If a name is specified, an additional column is generated under the name,
+        with the exact timestamp. Only with 'replace_index' == False
+    drop_original: bool, optional. Default is True.
+        Whether the base_time_field and time_offset_fields should be dropped in the
+        final DataFrame
+    handle_invalid: str, optional. Default is 'return_original'
```

Comment:
@wpreimes  Let me know if this makes sense for you; I believe it should be fine and flexible like this